src/EventListener/ReadListener.php

```diff
@@ -147,6 +148,10 @@ private function getSubresourceData(Request $request, array $attributes, array $
 
         $data = $this->subresourceDataProvider->getSubresource($attributes['resource_class'], $identifiers, $attributes['subresource_context'], $attributes['subresource_operation_name']);
 
+        if (null === $data) {
+            throw new ItemNotFoundException(sprintf('Object of type "%s" with identifiers %s could not be found', $attributes['resource_class'], implode(',', $identifiers)), 400);
```

Comment:
Why don't we use the `NotFoundHttpException`  here?

Reply:
I think it's better yeah.